### ISSUE TITLE
Kodi displays two InputStream Helper program add-ons, clicking right one fails.

### ISSUE THREAD

**Issue by mediaminister** [emilsvennesson/script.module.inputstreamhelper#108]
### Describe the bug ###

Kodi displays two InputStream Helper program add-ons in the add-on menu, clicking the right one fails.

### To Reproduce ###
1. Open Kodi
2. Select Add-ons
3. See a double program add-on item for InputStream Helper
4. Select and open the right item
6. You get InputStream Helper error. Check log...

### Expected behavior ###

There should be only a single program add-on item for InputStream Helper in the Kodi add-on menu. Clicking this item should open the InputStream Helper settings.

### Additional context ###

- Operating system: Windows 10
- Kodi version: Kodi 19 alpha with Python 3
- Addon version: master branch
- Using a VPN: no
- Country you are using the addon from: Belgium

### Log (if available) ###
```paste below
2019-08-28 20:26:37.342 T:10156   DEBUG: CPythonInvoker(2, C:\Users\test\AppData\Roaming\Kodi\addons\script.module.inputstreamhelper\addon.py): the source file to load is "C:\Users\test\AppData\Roaming\Kodi\addons\script.module.inputstreamhelper\addon.py"
2019-08-28 20:26:37.342 T:10156   DEBUG: CPythonInvoker(2, C:\Users\test\AppData\Roaming\Kodi\addons\script.module.inputstreamhelper\addon.py): setting the Python path to C:\Users\test\AppData\Roaming\Kodi\addons\script.module.inputstreamhelper;C:\Users\test\AppData\Roaming\Kodi\addons\script.module.inputstreamhelper\lib;C:\Users\test\AppData\Roaming\Kodi\addons\script.module.kodi-six\libs;C:\Users\test\AppData\Roaming\Kodi\addons\script.module.routing\lib;C:\Program Files\Kodi;C:\Program Files\Kodi\system\python\Lib;C:\Program Files\Kodi\kodi.zip;C:\Program Files\Kodi\system\python\bin\python\Lib;C:\Program Files\Kodi\system\python\bin\python\Lib\lib-dynload;C:\Program Files\Kodi\system\python\bin\python\Lib\lib-dynload\RelWithDebInfo;C:\Program Files\Kodi\system\python\bin\python\Lib\plat-win32;C:\Program Files\Kodi\system\python;C:\Program Files\Kodi\system\python\lib\site-packages
2019-08-28 20:26:37.343 T:10156   DEBUG: CPythonInvoker(2, C:\Users\test\AppData\Roaming\Kodi\addons\script.module.inputstreamhelper\addon.py): entering source directory C:\Users\test\AppData\Roaming\Kodi\addons\script.module.inputstreamhelper
2019-08-28 20:26:37.343 T:10156   DEBUG: CPythonInvoker(2, C:\Users\test\AppData\Roaming\Kodi\addons\script.module.inputstreamhelper\addon.py): instantiating addon using automatically obtained id of "script.module.inputstreamhelper" dependent on version 2.25.0 of the xbmc.python api
2019-08-28 20:26:38.458 T:10156   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <class 'IndexError'>
                                            Error Contents: list index out of range
                                            Traceback (most recent call last):
                                              File "C:\Users\test\AppData\Roaming\Kodi\addons\script.module.inputstreamhelper\addon.py", line 9, in <module>
                                                plugin = Plugin()
                                              File "C:\Users\test\AppData\Roaming\Kodi\addons\script.module.routing\lib\routing.py", line 57, in __init__
                                                self.handle = int(sys.argv[1]) if sys.argv[1].isdigit() else -1
                                            IndexError: list index out of range
                                            -->End of Python script error report<--

```

**Comment by dagwieers** [comment]
Yes, it requires a fix from **script.module.routing** which is fixed at: https://github.com/tamland/kodi-plugin-routing/pull/16

The fact we have two entries is unfortunate, but cannot be easily solved.

**Comment by dagwieers** [comment]
The reason is hidden in addon.xml:
```xml
    <!-- This is needed for running from RunPlugin() -->
    <extension point="xbmc.python.pluginsource" library="addon.py">
        <provides>executable</provides>
    </extension>
    <!-- This is needed for running from Information pane -->
    <extension point="xbmc.python.script" library="addon.py"/>
```

**Comment by mediaminister** [comment]
Do we really need the extension point `xbmc.python.pluginsource` and is this even allowed for a Kodi script module that is submitted to https://github.com/xbmc/repo-scripts

**Comment by dagwieers** [comment]
If we don't have it, the information pane has a `Run` button that gives an error or doesn't work IIRC.

**Comment by mediaminister** [comment]
Why do we have the following line in addon.xml
```xml
    <extension point="xbmc.python.module" library="lib"/>
```

**Comment by mediaminister** [comment]
If we remove `<provides>executable</provides>`, we only have one item, but the crashing item.

**Comment by dagwieers** [comment]
If we don't have **xbmc.python.module** it is no longer importable by other add-ons (i.e. the module will not be in PYTHONPATH).

**Comment by dagwieers** [comment]
> If we remove <provides>executable</provides>, we only have one item, but the crashing item.

 Can you still run the API?

**Comment by mediaminister** [comment]
Yes, 
I added this line to autoexec.py:
```
xbmc.executebuiltin('RunPlugin("plugin://script.module.inputstreamhelper/widevine/install")')```
But I get an error (testing on Linux now)
```
2019-08-28 21:33:56.766 T:139972873287424   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.IOError'>
                                            Error Contents: [Errno 2] No such file or directory: u'/home/test/.kodi/cdm/libwidevinecdm.so'
                                            Traceback (most recent call last):
                                              File "/home/test/.kodi/addons/script.module.inputstreamhelper/addon.py", line 40, in <module>
                                                plugin.run(sys.argv)
                                              File "/home/test/.kodi/addons/script.module.routing/lib/routing.py", line 115, in run
                                                self._dispatch(path)
                                              File "/home/test/.kodi/addons/script.module.routing/lib/routing.py", line 126, in _dispatch
                                                view_func(**kwargs)
                                              File "/home/test/.kodi/addons/script.module.inputstreamhelper/addon.py", line 30, in widevine_install
                                                Helper('mpd', drm='widevine').install_widevine()
                                              File "/home/test/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 681, in install_widevine
                                                return self._install_widevine_x86()
                                              File "/home/test/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 579, in _install_widevine_x86
                                                self._unzip(self._ia_cdm_path())
                                              File "/home/test/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 849, in _unzip
                                                zip_obj.extractall(unzip_dir)
                                              File "/usr/lib/python2.7/zipfile.py", line 1063, in extractall
                                                self.extract(zipinfo, path, pwd)
                                              File "/usr/lib/python2.7/zipfile.py", line 1051, in extract
                                                return self._extract_member(member, path, pwd)
                                              File "/usr/lib/python2.7/zipfile.py", line 1106, in _extract_member
                                                file(targetpath, "wb") as target:
                                            IOError: [Errno 2] No such file or directory: u'/home/test/.kodi/cdm/libwidevinecdm.so'
                                            -->End of Python script error report<--
```

**Comment by mediaminister** [comment]
The above error happens when running `plugin://script.module.inputstreamhelper/widevine/install` with version 0.4.0 while the Kodi cdm directory has still the layout of version 0.3.5

**Comment by dagwieers** [comment]
I can confirm that removing `<provides>executable</provides>` works, you get a disabled `Run` button, but that doesn't matter, we only need the `Settings` button.

**Comment by dagwieers** [comment]
@mediaminister Maybe open a new issue for this one. We want to fix this for v0.4.0,if possible.

**Comment by mediaminister** [comment]
When upgrading from 0.3.5 to 0.4.0 you don't get this error. It just happens if you remove 0.3.5 and choose to remove all settings an then install 0.4.0 and then run the install function.
I'll create a new issue for this.
